test-loop-tests/src/tests/spice.rs

```diff
@@ -158,8 +158,159 @@ fn test_spice_chain() {
         .shutdown_and_drain_remaining_events(Duration::seconds(20));
 }
 
+#[cfg(feature = "test_features")]
+#[test]
+#[cfg_attr(not(feature = "protocol_feature_spice"), ignore)]
+fn test_spice_chain_with_missing_chunks() {
+    use crate::utils::account::{
+        create_validators_spec, rpc_account_id, validators_spec_clients_with_rpc,
+    };
+    use crate::utils::node::TestLoopNode;
+
+    init_test_logger();
+    let accounts: Vec<AccountId> =
+        (0..100).map(|i| format!("account{}", i).parse().unwrap()).collect_vec();
+
+    // With 2 shards and 4 producers we should still be able to include all transactions
+    // when one of the producers starts missing chunks.
+    let num_producers = 4;
+    let num_validators = 0;
+    let shard_layout =
+        ShardLayout::multi_shard_custom(vec![accounts[accounts.len() / 2].clone()], 1);
```

Comment:
just a sanity check: do we really need 100/2=50 shards? 🤔

Reply:
It's 2 shards. I restructured the code in 9c0ab7dcd to make it clearer that boundary_account is passed in.